grafana/grafana
# InfluxDB: Fix template variable interpolation

**What is this feature?**

Template variables that have special characters (`/`, `//`, `)`, etc) are causing interpolation errors. This is fixing issues by using built-in `regex` interpolation formatter.

**Why do we need this feature?**

Better template variable interpolation

**Who is this feature for?**

InfluxDB influxql users

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/77465
Fixes https://github.com/grafana/grafana/issues/80003
Fixes https://github.com/grafana/grafana/issues/79692
Fixes https://github.com/grafana/grafana/issues/80242
Fixes https://github.com/grafana/grafana/issues/79765
Fixes https://github.com/grafana/grafana/issues/81533

**Special notes for your reviewer:**

### How to test?
- Create a template variable that returns values with special characters
- Set that variable as "multi-value" and "include all option" 
- i.e. `SHOW TAG VALUES WITH KEY="path"`
- Use that variable in your queries with raw query and visual query builder
- Select multiple values or "All"
- With the fix it should work as normal
- Use the variable in other variable query, this should work too. 

Please check that:
